lib/uwb/include/uwb/UwbSession.hxx

```diff
@@ -82,33 +82,48 @@ public:
 
     /**
      * @brief Set the type of mac address to be used for session participants.
-     * 
+     *
      * @param uwbMacAddressType The type of mac address to use.
      */
     void
     SetMacAddressType(UwbMacAddressType uwbMacAddressType) noexcept;
 
     /**
      * @brief Add a peer to this session.
-     * 
+     *
      * @param peerMacAddress The mac address of the peer. This is expected to be
      * in the mac address format configured for the session.
      */
     void
     AddPeer(UwbMacAddress peerMacAddress);
 
     /**
-     * @brief 
+     * @brief
      */
     void
     StartRanging();
 
     /**
-     * @brief 
+     * @brief
      */
     void
     StopRanging();
 
+    /**
+     * @brief Set the Session Status object
+     *
+     */
+    void
+    SetSessionStatus(const uwb::protocol::fira::UwbSessionStatus&);
+
+    /**
+     * @brief Get the Session Status object
+     *
+     * @return const UwbSessionStatus&
+     */
+    const uwb::protocol::fira::UwbSessionStatus&
+    GetSessionStatus();
```

Comment:
Is there any reason we need this right now? If there's no specific use, then let's remove it, especially since returning a reference to an object that can be asynchronously updated isn't thread safe. We can solve that problem, but I'd rather not introduce complexity now if we don't need it.

Reply:
Generally speaking I like to make a getter for every setter but I agree that we don't need it right now and I didn't think that the function could be asynchronously called.

Reply:
> Generally speaking I like to make a getter for every setter

I agree this is a good strategy in general.

> I didn't think that the function could be asynchronously called.

Here, the getter and setter can be called concurrently if the driver updates the session status while the service is trying to obtain the session status (say, to fulfill an API call).